src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -99,23 +149,155 @@ const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const
+        // -> Apparent wind angle, rad, negative to port
+        if (windAngle >M_PI) windAngle *= -1;
+        update->setEnvironmentWindAngleApparent(windAngle);
+        DEBUG("AWA: %d °",SKRadToDeg(windSpeed));
       }
     }
+    // We do not take True Wind Speed as we do not know how the system has calculated it!!
+    // Some instruments take SOG/COG then True Wind => Ground Wind
 
     _sku = update;
     return *_sku;
   }
   else {
-    INFO("Unable to parse N2kMsg with PGN %i", msg.PGN);
+    DEBUG("Unable to parse NMEA2000 with PGN %i", msg.PGN);
     return _invalidSku;
   }
 }
+
+//  ***********************************************
+//   PGN 127245 Rudder
+//        →  sid
+//        →  RudderPosition [rad]
+// *  ********************************************** */
+const SKUpdate& SKNMEA2000Parser::parse127245(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char instance;
+  tN2kRudderDirectionOrder rudderDirectionOrder;
+  double rudderPosition;
+  double angleOrder;
+
+  if (ParseN2kRudder(msg,rudderPosition,instance,rudderDirectionOrder,angleOrder) ) {
+    SKUpdateStatic<1> *update = new SKUpdateStatic<1>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+
+    if (!N2kIsNA(rudderPosition) && (rudderPosition < M_PI_2 / 2) && (rudderPosition > M_PI_2 / 2) ) {
+      // -> Current rudder angle, +ve is rudder to Starboard
+      update->setSteeringRudderAngle(rudderPosition);
+      DEBUG("Rudder: %d",SKRadToDeg(rudderPosition));
+    }
+
+    _sku = update;
+    return *_sku;
+  }
+  else {
+    DEBUG("Unable to parse NMEA2000 with PGN %i", msg.PGN);
+    return _invalidSku;
+  }
+}
+// *****************************************************************************
+//    PGN 127250 VESSEL HEADING RAPID
+//    Heading sensor value with a flag for True or Magnetic.
+//    Heading               Heading in radians
+//      - Deviation         Magnetic deviation in radians. Use N2kDoubleNA for undefined value.
+//      - Variation         Magnetic variation in radians. Use N2kDoubleNA for undefined value.
+//      tN2kHeadingReference
+//        N2khr_true=0,
+//        N2khr_magnetic=1
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse127250(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  tN2kHeadingReference headingReference;
+  double heading = N2kDoubleNA;
+  double deviation = N2kDoubleNA;
+  double variation = N2kDoubleNA;
+
+  if (ParseN2kHeading(msg,sid,heading,deviation,variation,headingReference) ) {
+
+    if (headingReference == N2khr_magnetic) {
+      // Deviation and Variation could be sent
+      // (Timo's NMEA Simulator is sending Dev and Var at magnetic HDG only)
+      heading += deviation + variation;
```

Comment:
I do not think we should transform the heading. If it's magnetic we should just pass the fact that it is magnetic but not transform it.

Reply:
OMG a bad BUG!!
Of course not!!!
(....now I know, what you meant in the discussion of the last PR....)

Reply:
I did an additionally validation check:

`if (!N2kIsNA(heading) && heading >= 0 && heading <= M_PI_2) {`

What do you think?